.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,38 @@
+name: Pull request build
+
+on:
+  pull_request:
+    branches:
+      - dev
+      - main
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      # Checkout the repository
+      - name: Checkout Code
+        uses: actions/checkout@v3
+
+      # Set up JDK
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+
+      # Install Ballerina
+      - name: Install Ballerina
```

Comment:
We have already specified the required Ballerina version in the `gradle.properties` file. Can you check how we can get that version here rather than hard-coding the version ?

Reply:
```
# Read Ballerina version from gradle.properties
- name: Read Ballerina Version
  id: ballerina_version
  run: echo "ballerina_version=$(grep ballerinaVersion gradle.properties | cut -d'=' -f2)" >> $GITHUB_ENV

# Install Ballerina
- name: Install Ballerina
  uses: ballerina-platform/setup-ballerina@v1
  with:
    version: ${{ env.ballerina_version }}
    ```
          
I think this should work.

Reply:
+1